lib/server/services/shift/ShiftService.js

```diff
@@ -64,6 +65,29 @@ class ShiftService {
             typeSpecific: null,
         };
 
+        // Find all EoS reports from the last 16 hours and sort by most recent first
+        const pastShift = { start: new Date() - SHIFT_DURATION * 2, end: new Date() };
```

Comment:
This is not correct, because here you will have shift that start exactly 16 hours ago, which is not the reality. If you run this code at 2PM you want `start` to be 23PM the day before, not 22PM for example.

Instead I would suggest to simply use the current shift, something like this should work:

```js
const pastShift = getShiftFromTimestamp(shift.start - SHIFT_DURATION);
```

You should extract this in a function such as `getPreviousShift` and add test to it, checking corner cases such as exact 3PM